celery/events/receiver.py

```diff
@@ -90,7 +90,8 @@ def capture(self, limit=None, timeout=None, wakeup=True):
         unless :attr:`EventDispatcher.should_stop` is set to True, or
         forced via :exc:`KeyboardInterrupt` or :exc:`SystemExit`.
         """
-        return list(self.consume(limit=limit, timeout=timeout, wakeup=wakeup))
+        for _ in self.consume(limit=limit, timeout=timeout, wakeup=wakeup):
```

Comment:
Although, the fix seems correct, I could not find a connection with worker functionality. Event capturing is performed by specific commands only. Could you please explain how this method affects the worker? Thanks!